File: solutions/Arrays & Hashing/238_product-of-array-except-self.py
from typing import List

class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        res = [1] * len(nums)
        acc = 1

        for i in range(len(nums)):
            res[i] *= acc
            acc *= nums[i]

        acc = 1
        for i in range(len(nums) -1, -1, -1):
            res[i] *= acc
            acc *= nums[i]

        return res

File: C++/Arrays & Hashing/238_product-of-array-except-self.cpp
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        size_t n = nums.size();
        vector<int> res(n, 1);

        int acc = 1;
        for (int i = 0; i < n; ++i) {
            res[i] *= acc;
            acc *= nums[i];
        }

        acc = 1;
        for (int i = n - 1; i >= 0; --i) {
            res[i] *= acc;
            acc *= nums[i];
        }

        return res;
    }
};